File: app/src/main/java/com/kfang/fencemelibrary/presentation/MainContract.java
package com.kfang.fencemelibrary.presentation;

import com.kfang.fencemelibrary.model.Fencer;

/**
 * Main Contract class
 */

public class MainContract {
    public interface MainPresenter {
        Fencer getRedFencer();

        Fencer getGreenFencer();

        void setTieBreaker(boolean status);

        void handleCarding(String cardingPlayer, String cardToGive);

        void toggleTimer();

        void startTimer();

        void stopTimer();

        void resetBout();

        void resetScores();

        void setTimer(int time);

        void resetTimer();

        boolean equalPoints();

        boolean checkForVictories(Fencer fencer);

        Fencer checkForVictories();

        void resetCards();

        Fencer randomFencer();

        Fencer higherPoints();

        void incrementBothPoints();

        boolean timerRunning();

        boolean getTiebreaker();

        int getCurrentSeconds();

        int getBoutLengthMinutes();

        boolean vibrateOnTimerFinish();

        boolean stayAwakeDuringTimer();

        int getPointsToWin();

        boolean popupOnScoreChange();

        boolean restoreOnAppReset();

        boolean enableDoubleTouch();

        boolean vibrateOnTimerToggle();

        boolean pauseOnScoreChange();
    }

    public interface MainView {
        void updateTime(String time);

        void enableChangingScore();

        void enableTimerButton();

        void disableTimerButton();

        void disableChangingScore();

        void timerUp();

        void updateToggle(String colorTo, int text);

        void displayWinnerDialog(Fencer winner);

        void vibrateTimer();
    }

    public interface FenceTimer {
        void startTimer();

        void stopTimer();

        void setTimer(int time);

        int getTime();
    }

}

File: app/src/main/java/com/kfang/fencemelibrary/presentation/MainContract.kt
package com.kfang.fencemelibrary.presentation

import com.kfang.fencemelibrary.model.Fencer

/**
 * Main Contract class
 */

class MainContract {
    interface MainPresenter {
        val redFencer: Fencer

        val greenFencer: Fencer

        fun handleCarding(cardingPlayer: String, cardToGive: String)

        fun toggleTimer()

        fun startTimer()

        fun stopTimer()

        fun resetBout()

        fun resetScores()

        fun setTimerSeconds(seconds: Int)

        fun resetTimer()

        fun equalPoints(): Boolean

        fun checkForVictories(fencer: Fencer): Boolean

        fun checkForVictories(): Fencer?

        fun resetCards()

        fun randomFencer(): Fencer

        fun higherPoints(): Fencer?

        fun incrementBothPoints()

        var timerRunning: Boolean

        var tiebreaker: Boolean

        var currentDeciSeconds: Int

        var boutLengthMinutes: Int

        var sabreMode: Boolean

        fun vibrateOnTimerFinish(): Boolean

        fun stayAwakeDuringTimer(): Boolean

        var pointsToWin: Int

        fun popupOnScoreChange(): Boolean

        fun restoreOnAppReset(): Boolean

        fun enableDoubleTouch(): Boolean

        fun vibrateOnTimerToggle(): Boolean

        fun volumeButtonTimerToggle(): Boolean

        fun pauseOnScoreChange(): Boolean

        fun toggleSabreMode()
    }

    interface MainView {

        fun hideTimer()

        fun showTimer()

        fun updateTime(time: String)

        fun setScoreChangeability(allowChange: Boolean)

        fun timerUp()

        fun displayWinnerDialog(winner: Fencer)

        fun vibrateStart()

        fun vibrateStop()

        fun setTimerColor(color: String)
    }

    interface FenceTimer {
        fun startTimer()

        fun stopTimer()

        fun setTimerSeconds(seconds: Int)

        var deciSeconds: Int
    }

}
